### PR TITLE
Initial monorepo setup

### LINKED ISSUE #6: Monorepo Setup
**Is your feature request related to a problem? Please describe.**
As a developer I would like an easy way to manage all the individual packages in this repo.

**Describe the solution you'd like**
A [Lerna](https://github.com/lerna/lerna) monorepo will improve the developer experience by simplifying project setup.

We could have two packages:
```
/backend
/app
```

Furthermore, the root level monorpeo should contain any pre-commit hooks, such as for linting.

**Describe alternatives you've considered**
- No monorepo system, could just have two packages. 
- [nx](https://nx.dev/), but might be a lot of setup and boilerplate.

### PR DESCRIPTION
**Pull request by frasermcc9**
# Description

This addresses #6 by implementing a basic Lerna monorepo.

From `CONTRIBUTING.md`:

Setup

This project uses a Lerna Monorepo. Lerna is a tool used to manage multiple JavaScript packages. There are two self-descriptive packages, the backend package and the frontend package.

To Setup

1. At the root level of the project, run `npm i`. This will install the packages required for Lerna to work, as well as automatically install packages for the frontend and backend packages.

2. Once again at the project root, run `npm run start`. This will automatically run the `start` command in all nested packages, which will simultaneously run the frontend and backend.

## Type of change

- [X] **New feature** (non-breaking change which adds functionality)

# Checklist:
Leave blank if not applicable

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by j-chad** [comment]
can you add documentation please

**Comment by frasermcc9** [comment]
Added to the contributing file

**Comment by fishmonger45** [comment]
Are we set on using npm instead of yarn? yarn has better dependency management which just helps in the long run (lockfiles are actually frozen etc). I'm not particularly deep on webdev so let me know if we should actually care or if i'm wrong

**Comment by frasermcc9** [comment]
I think nowadays the tools are basically the same. NPM can do proper locking if you use `npm ci` instead of `npm i`. 

I'm fine if we want to switch to yarn, but probably not going to use yarn workspaces since I doubt we are going to have any additional packages depending on each other.

**Comment by j-chad** [comment]
We should also add the setup guide on README

**Comment by frasermcc9** [comment]
Switched to yarn and added a quickstart to the readme (which points to the contributing file with more detailed instructions).

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 # Please add IDE and OS specific files in a global gitignore on your system
+
+node_modules
```
Comment (j-chad) [inline .gitignore:3]: might want to add https://www.toptal.com/developers/gitignore/api/node. I don't want to have people keep committing these files accidentally

**Review by j-chad — APPROVED** [review]
Would like to have some more files added to gitignore, otherwise LGTM

**Pull request merged**